backend/src/apiserver/storage/experiment_store.go

```diff
@@ -388,7 +362,15 @@ func (s *ExperimentStore) ArchiveExperiment(expId string) error {
 			"Failed to archive experiment %s. error: '%v'", expId, err.Error())
 	}
 
-	_, err = tx.Exec(updateRunsSql, updateRunsArgs...)
+	var arguments []interface{}
+	arguments = append(arguments, model.StorageStateArchived.ToString(), model.RunResourceType, expId, model.ExperimentResourceType)
+	query := s.db.UpdateWithJointOrFrom(
+		"run_details",
+		"resource_references",
+		"StorageState = ?",
+		"run_details.UUID = resource_references.ResourceUUID",
+		"resource_references.ResourceType = ? AND resource_references.ReferenceUUID = ? AND resource_references.ReferenceType = ?")
+	_, err = tx.Exec(query, arguments...)
```

Comment:
This could be a little more readable if we limit the scope of `UpdateWithJointOrFrom` to only `UPDATE %s INNER JOIN %s ON %s`:

```golang
query, args, err := s.db.UpdateWithJointOrFrom(
		"run_details",
		"resource_references",
                "run_details.UUID = resource_references.ResourceUUID").
                SetMap(...).
                Where(...).
                ToSql()
```

Reply:
That is a great idea, but at least for now, the implementation does not look trivial to me. Squirrel heavily uses `builder` design pattern, so to be able to call `SetMap` and `Where` - IMO t I will need to extend squirrel library - by creating a new `StatementBuilder`, which still needs to build different requests based if MySQL or SQLite ... Let me think

Reply:
That makes sense. Thank you. Perhaps another alternative is to opt for gorm v2.